adapters/thetradedesk/thetradedesk.go

```diff
@@ -232,3 +235,12 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 		templateEndpoint:       template,
 	}, nil
 }
+
+func resolveAuctionPriceMacros(bid *openrtb2.Bid) {
+	if bid == nil {
+		return
+	}
+	price := strconv.FormatFloat(bid.Price, 'f', -1, 64)
+	bid.NURL = strings.Replace(bid.NURL, "${AUCTION_PRICE}", price, -1)
+	bid.AdM = strings.Replace(bid.AdM, "${AUCTION_PRICE}", price, -1)
+}
```

Comment:
The unit test coverage looks good but our position is that all code coverage that can be achieved via the JSON test framework should be done that way. As a result, please add a supplemental JSON test that verifies the macros are replaced correctly.